Library/Homebrew/language/node.rb

```diff
@@ -4,15 +4,23 @@ def self.npm_cache_config
       "cache=#{HOMEBREW_CACHE}/npm_cache\n"
     end
 
-    def self.pack_for_installation
+    def self.pack_for_installation(prepublish_requires_deps)
+      # Some packages are requiring (dev)dependencies to be in place when
+      # running the prepublish script (which is run before pack). In this case
+      # we have to install all dependencies a first time already before
+      # executing npm pack (and a second time when doing the actual install).
+      safe_system "npm", "install" if prepublish_requires_deps
+
       # Homebrew assumes the buildpath/testpath will always be disposable
       # and from npm 5.0.0 the logic changed so that when a directory is
       # fed to `npm install` only symlinks are created linking back to that
       # directory, consequently breaking that assumption. We require a tarball
       # because npm install creates a "real" installation when fed a tarball.
-      output = Utils.popen_read("npm pack").chomp
-      raise "npm failed to pack #{Dir.pwd}" unless $CHILD_STATUS.exitstatus.zero?
-      output
+      output = Utils.popen_read("npm pack")
+      unless $CHILD_STATUS.exitstatus.zero? && !output.lines.empty?
+        raise "npm failed to pack #{Dir.pwd}"
+      end
+      output.lines.last.chomp
```

Comment:
This change is needed because npm pack could contain verbose output from a prepublish script and we are only interested in the last line from npm itself containing the name of the created package.

Does someone with more knowledge in ruby has a suggestion how this could be improved?